### PR TITLE
Re-implementation in python

### PR DESCRIPTION
**Pull request by TheDataLeek**
Woo

### CODE REVIEW

File: todo_tools.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    else:
+        sys.stdin = open('/dev/tty')
+        run_as_hook(args)
+
+
+def run_as_checker(args):
+    outstanding = []
+    with open(args.file, 'r') as todofile:
+        for line in todofile.readlines():
+            if dt.today() > dt.strptime(line[:10], '%Y-%m-%d'):
```
Comment (sergeio) [inline todo_tools.py:55]: This will start failing in 8000 years.  (Joke.  Don't change it ; )

File: todo_tools.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import argparse
+import re
+from datetime import datetime as dt
+import time
+
+from fabulous import color
+import git
+
+# TODO(@zeb[2016-08-04]): pls check if werks
+# TODO(@zeb[2015-08-04]): This should fail
```
Comment (sergeio) [inline todo_tools.py]: You can probably get rid of these now?

File: todo_tools.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+# 
+
+"""
+
+from __future__ import print_function
+
+import sys
+import os
+import shutil
```
Comment (sergeio) [inline todo_tools.py]: nit:  Sorting these would be nice : D

File: todo_tools.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+For each repository you want to log TODO's, run
+
+./todo-tools.py -i ~/myawesomerepo
+
+# 
+
+"""
+
+from __future__ import print_function
```
Comment (sergeio) [inline todo_tools.py:19]: Yay python 3!

File: parse_new_todos [hunk 5]
```diff
@@ -104,3 +104,5 @@ difiles() {
 }
 
 main $*
+
+# test
```
Comment (sergeio) [inline parse_new_todos]: Can probably get rid of this

File: todo_tools.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+def main():
+    args = get_args()
+
+    if sys.version_info[0] == 2:
+        input = raw_input
+
+    if args.install and os.path.isdir(os.path.join(args.install, '.git')):
+        # TODO: copy instance of running file to given directory
+        shutil.copy('./todo_tools.py', os.path.join(args.install, '.git/hooks/post-commit'))
```
Comment (sergeio) [inline todo_tools.py]: This line might be too long?

**Comment by sergeio** [comment]
This is great.  Before I merge, do you think you could do some profiling?  

Because we want to use this in a post-commit hook with Khan/webapp, and committing already takes a long time, we're really sensitive to anything that'll make that go up even further.

Maybe try running this on the last 1000 - 2000 commits, and get some stats / graphs.
https://phabricator.khanacademy.org/rKL9b3abf8cff2606fd60ed3dcd58492930390b2ef8 is a commit with a lot of files changed.

I'd also look for a commit with really really long files and make sure it runs well there too.

Again, awesome stuff.  Thanks for doing this!

**Comment by TheDataLeek** [comment]
Hey @sergeio if you take a look at that most recent commit, I added profiling as well as a histogram that shows the results. TL;DR when examining the last 1000 commits, the vast majority of them take less than a tenth of a second to analyze, with the max time around 1-2 seconds.

Lemme know how that looks.

**Comment by rileyjshaw** [comment]
Nice!!

**Pull request merged**